Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1219,35 +1224,52 @@ void CGContextSetCharacterSpacing(CGContextRef context, CGFloat spacing) {
 }
 
 /**
- @Status Interoperable
+ @Status Caveat
+ @Notes Only kCGTextFill is supported.
 */
 void CGContextSetTextDrawingMode(CGContextRef context, CGTextDrawingMode mode) {
     NOISY_RETURN_IF_NULL(context);
-    UNIMPLEMENTED();
+    auto& state = context->CurrentGState();
+    state.textDrawingMode = mode;
 }
 
 /**
  @Status Interoperable
 */
 void CGContextSetFont(CGContextRef context, CGFontRef font) {
     NOISY_RETURN_IF_NULL(context);
-    UNIMPLEMENTED();
+    NOISY_RETURN_IF_NULL(font);
+    auto& state = context->CurrentGState();
+    state.font = font;
 }
 
 /**
  @Status Interoperable
+ @Notes encoding is not used as Windows fonts do not support the MacRoman encoding.
 */
 void CGContextSelectFont(CGContextRef context, const char* name, CGFloat size, CGTextEncoding encoding) {
     NOISY_RETURN_IF_NULL(context);
-    UNIMPLEMENTED();
+
+    auto fontName = woc::MakeStrongCF<CFStringRef>(CFStringCreateWithCString(nullptr, name, kCFStringEncodingUTF8));
```

Comment:
sure, I feel like this could lead to every single instance of woc:MakeStrong/ woc:* end up being auto.
your call.

---
In reply to: [96745127](https://github.com/Microsoft/WinObjC/pull/1729#discussion_r96745127) [](ancestors = 96745127)